packages/controllers/src/snaps/json-schemas/validateNpmSnapPackageJson.js

```diff
@@ -1 +1 @@
ject"};if(vErrors === null){vErrors = [err16];}else {vErrors.push(err16);}errors++;}}if(data.repository !== undefined){let data7 = data.repository;if(data7 && typeof data7 == "object" && !Array.isArray(data7)){if(data7.type === undefined){const err17 = {instancePath:instancePath+"/repository",schemaPath:"#/properties/repository/required",keyword:"required",params:{missingProperty: "type"},message:"must have required property '"+"type"+"'"};if(vErrors === null){vErrors = [err17];}else {vErrors.push(err17);}errors++;}if(data7.url === undefined){const err18 = {instancePath:instancePath+"/repository",schemaPath:"#/properties/repository/required",keyword:"required",params:{missingProperty: "url"},message:"must have required property '"+"url"+"'"};if(vErrors === null){vErrors = [err18];}else {vErrors.push(err18);}errors++;}for(const key0 in data7){if(!((key0 === "type") || (key0 === "url"))){const err19 = {instancePath:instancePath+"/repository",schemaPath:"#/properties/repository/additionalProperties",keyword:"additionalProperties",params:{additionalProperty: key0},message:"must NOT have additional properties"};if(vErrors === null){vErrors = [err19];}else {vErrors.push(err19);}errors++;}}if(data7.type !== undefined){let data8 = data7.type;if(typeof data8 === "string"){if(func8(data8) < 1){const err20 = {instancePath:instancePath+"/repository/type",schemaPath:"#/properties/repository/properties/type/minLength",keyword:"minLength",params:{limit: 1},message:"must NOT have fewer than 1 characters"};if(vErrors === null){vErrors = [err20];}else {vErrors.push(err20);}errors++;}}else {const err21 = {instancePath:instancePath+"/repository/type",schemaPath:"#/properties/repository/properties/type/type",keyword:"type",params:{type: "string"},message:"must be string"};if(vErrors === null){vErrors = [err21];}else {vErrors.push(err21);}errors++;}}if(data7.url !== undefined){let data9 = data7.url;if(typeof data9 === "string"){if(func8(data9) < 1){const err22 = {instancePath:instancePath+"/repository/url",schemaPath:"#/properties/repository/properties/url/minLength",keyword:"minLength",params:{limit: 1},message:"must NOT have fewer than 1 characters"};if(vErrors === null){vErrors = [err22];}else {vErrors.push(err22);}errors++;}}else {const err23 = {instancePath:instancePath+"/repository/url",schemaPath:"#/properties/repository/properties/url/type",keyword:"type",params:{type: "string"},message:"must be string"};if(vErrors === null){vErrors = [err23];}else {vErrors.push(err23);}errors++;}}}else {const err24 = {instancePath:instancePath+"/repository",schemaPath:"#/properties/repository/type",keyword:"type",params:{type: "object"},message:"must be object"};if(vErrors === null){vErrors = [err24];}else {vErrors.push(err24);}errors++;}}}else {const err25 = {instancePath,schemaPath:"#/type",keyword:"type",params:{type: "object"},message:"must be object"};if(vErrors === null){vErrors = [err25];}else {vErrors.push(err25);}errors++;}validate20.errors = vErrors;return vErrors;}
```

Comment:
Why do we include this in repository instead of generating those at build time

Reply:
We'll probably stop checking these in and move the generation to a `postinstall` step in the near future.